slick-codegen/src/main/scala/slick/codegen/AbstractGenerator.scala

```diff
@@ -211,16 +214,20 @@ abstract class AbstractGenerator[Code,TermName,TypeName](model: m.Model)
       /**
        * Underlying Scala type of this column.
        * Override this to just affect the data type but preserve potential Option-wrapping.
-       * Override GeneratorHelpers#mapJdbcTypeString for generic adjustments.
        * @group Basic customization overrides
        */
       def rawType: Code = parseType(model.tpe)
+      /**
+       * Indicates whether the exposed type of this column should be wrapped in an Option. Useful for automatically created columns.
+       * @group Basic customization overrides
+       */
+      def exposeOption = false
       /** Possibly Option-wrapped Scala type of this column. @see rawType and @see exposedType */
       final def actualType: Code      = if(model.nullable) optionType(rawType) else rawType
-      /** Option of actualType if fakeNullable else actualType. Useful to expose autoInc columns as nullable. */
+      /** Option of actualType if fakeNullable else actualType. */
       final def exposedType: Code  = if(fakeNullable) optionType(actualType) else actualType
-      /** Indicates whether this column should be user facing as a nullable column with default None even though it is not. Useful for autoInc columns. */
-      final def fakeNullable = autoIncLastAsOption && autoInc
+      /** Indicates whether this column should be user facing as a nullable column with default None even though it is not. Useful for autoInc and automatically created columns. */
+      final def fakeNullable = exposeOption || (autoIncAsOption && autoInc)
```

Comment:
should we just make this non-final instead of adding a new field `exposeOption` ?

Maybe fakeOption is a better name than fakeNullable, now that we are basically on Scala-level, not on SQL-model level anymore.